### PR TITLE
feat: add action to generate docker images

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This generates images for the client and the api.  There are a few changes we could make (see below), but this has the basic functionality.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
Considerations:

- [ ] move the run scripts into separate .sh files
- [ ] create a dev docker-compose.yml 
- [ ] create dev Dockerfiles

following on from @nhcarrigan's suggestion https://github.com/freeCodeCamp/freeCodeCamp/pull/41187#issuecomment-819039369

At the moment this would be 'nice to have'.  The last time we had a Docker development environment the experience was not great, judging by the number of support requests we got.  Before we do this we need to figure out if it will need an entirely separate set of Dockerfiles and docker-compose.ymls (probably) and if we will be able to maintain it properly.

File: .github/workflows/docker-images.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: Generate Docker Images
+# TODO: it's triggered on PRs for QA, but it'll be on pushes to master when it's
+# done.
```
Comment (nhcarrigan) [inline .github/workflows/docker-images.yml:3]: Quick question - is this for deployment or for development? Where do the images go?

Comment (ojeytonwilliams) [reply .github/workflows/docker-images.yml:3]: It's for deployment.  The artifact should be available through [GitHub itself](https://github.com/actions/upload-artifact#where-does-the-upload-go), but the intention is to link it to whatever we use for deployment (TBD).

Comment (ojeytonwilliams) [reply .github/workflows/docker-images.yml:3]: [Tada!](https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/767447858) (at the bottom)

Comment (nhcarrigan) [reply .github/workflows/docker-images.yml:3]: Nice!!!

So my next question is - do we want this to run on merge to `main` or on merge to our production branches?

Comment (ojeytonwilliams) [reply .github/workflows/docker-images.yml:3]: I'm not 100% sure, but probably on merges to `main`.  As I see it the deployment sequence will end up being something like this:

1. build images, triggered by merge to `main`
2. run CI tests using those images
3. if they pass, push those images to staging VMs
4. if those are okay, repeat for production VMs

perhaps we'll still want to use production branches with this, but it's a different flow, so maybe not.

File: .github/workflows/docker-images.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          --build-arg FORUM_LOCATION=https://forum.freecodecamp.org \
+          --build-arg NEWS_LOCATION=https://www.freecodecamp.org/news \
+          --build-arg CLIENT_LOCALE=english \
+          --build-arg CURRICULUM_LOCALE=english \
+          --build-arg SHOW_UPCOMING_CHANGES=false \
+          --build-arg ALGOLIA_APP_ID=app_id_from_algolia_dashboard \
+          --build-arg ALGOLIA_API_KEY=api_key_from_algolia_dashboard \
+          --build-arg PAYPAL_CLIENT_ID=id_from_paypal_dashboard \
+          --build-arg DEPLOYMENT_ENV=staging \
+          -t fcc_client client.Dockerfile .
```
Comment (ojeytonwilliams) [inline .github/workflows/docker-images.yml:34]:
```suggestion
          -t fcc_client -f client.Dockerfile .
```

**Comment by ojeytonwilliams** [comment]
@raisedadead just so I don't forget, there can be an issue with `docker-compose` due to a lack of entropy https://github.com/docker/compose/issues/6678 I've run into this a bunch of times on DO droplets, without having a clue what was happening.  Turns out VMs, particularly ones that have just turned on, don't have enough entropy built up.  

We can install `rng-tools` and that alone seems to generate enough noise.  Any idea if Azure VMs do this by default?

**Comment by raisedadead** [comment]
Sure. I will keep that in mind.

**Review by raisedadead — COMMENTED** [review]
Thanks for your work on this @ojeytonwilliams - we are going to be using all of this logic soon. Closing for now, I am going to come with a fresh PR with you as the co-author.

**Pull request closed without merging**